suite-native/tokens/src/ethereumTokensSelectors.ts

```diff
@@ -171,27 +162,39 @@ export const selectAccountOrTokenAccountTransactions = (
     ) as WalletAccountTransaction[];
 };
 
+export const selectUniqueEtheruemTokens = (state: AccountsRootState & DeviceRootState) => {
+    const accounts = selectVisibleDeviceAccounts(state);
+
+    return pipe(
+        accounts,
+        A.filter(account => isEthereumAccountSymbol(account.symbol)),
+        A.map(account => account.tokens?.map(token => token.contract)),
+        A.flat,
+        // Don't use A.uniq(By) because it's slow for large arrays
+        t => Array.from(new Set(t)),
+        A.filter(tokenAddress => !!tokenAddress),
+    ) as TokenAddress[];
+};
+
 export const selectEthereumAccountsTokensWithFiatRates = memoizeWithArgs(
```

Comment:
now it does not check the fiat rates but token definitons, right? What about `selectKnownEthereumAccountTokens`? Or something similar.

Reply:
Good idea

Reply:
I will not rename it in this PR in the end. I am already playing with it in next PR for some selectors optimalizations.